notebooks/ch15/bias_cv.md

```diff
@@ -0,0 +1,498 @@
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+
+    .dataframe thead th {
+        text-align: right;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>Validation Error</th>
+    </tr>
+    <tr>
+      <th>Degree</th>
+      <th></th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>1</th>
+      <td>0.324820</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>0.045060</td>
+    </tr>
+    <tr>
+      <th>3</th>
+      <td>0.045418</td>
+    </tr>
+    <tr>
+      <th>4</th>
+      <td>0.045282</td>
+    </tr>
+    <tr>
+      <th>5</th>
+      <td>0.046272</td>
+    </tr>
+    <tr>
+      <th>6</th>
+      <td>0.046715</td>
+    </tr>
+    <tr>
+      <th>7</th>
+      <td>0.047140</td>
+    </tr>
+    <tr>
+      <th>8</th>
+      <td>0.047540</td>
+    </tr>
+    <tr>
+      <th>9</th>
+      <td>0.048055</td>
+    </tr>
+    <tr>
+      <th>10</th>
+      <td>0.047805</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+We can see that as we use higher degree polynomial features, the validation error decreases and increases again.
+
+
+```python
+# HIDDEN
+plt.figure(figsize=(10, 5))
+
+plt.subplot(121)
+plt.plot(cv_df.index, cv_df['Validation Error'])
+plt.scatter(cv_df.index, cv_df['Validation Error'])
+plt.title('Validation Error vs. Polynomial Degree')
+plt.xlabel('Polynomial Degree')
+plt.ylabel('Validation Error');
+
+plt.subplot(122)
+plt.plot(cv_df.index, cv_df['Validation Error'])
+plt.scatter(cv_df.index, cv_df['Validation Error'])
+plt.ylim(0.044925, 0.05)
+plt.title('Zoomed In')
+plt.xlabel('Polynomial Degree')
+plt.ylabel('Validation Error')
+
+plt.tight_layout();
+```
+
+
+![png](bias_cv_files/bias_cv_25_0.png)
+
+
+Examining the validation errors reveals that the most accurate model only used degree 2 polynomial features. Thus, we select the degree 2 polynomial model as our final model and fit it on the all of the training data at once. Then, we compute its error on the test set.
+
+
+```python
+# HIDDEN
+best_trans = transformers[1]
+best_model = LinearRegression(fit_intercept=False).fit(X_train_polys[1], y_train)
+
+training_error = mse_cost(best_model.predict(X_train_polys[1]), y_train)
+validation_error = cross_validation_errors[1]
+test_error = mse_cost(best_model.predict(best_trans.transform(X_test)), y_test)
+
+print('Degree 2 polynomial')
+print(f'  Training error: {training_error:0.5f}')
+print(f'Validation error: {validation_error:0.5f}')
+print(f'      Test error: {test_error:0.5f}')
+```
+
+    Degree 2 polynomial
+      Training error: 0.04409
+    Validation error: 0.04506
+          Test error: 0.04698
+    
+
+For future reference, `scikit-learn` has a [`cross_val_predict`](http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_val_predict.html) to automatically perform cross-validation, so we don't have to break the data into training and validation sets yourself. 
```

Comment:
From last time:
First sentence: "has a cross_val_predict [method]"
"ourselves" instead of "yourself"

Reply:
fixed